src/ngraph/node.hpp

```diff
@@ -531,32 +526,21 @@ namespace ngraph
         std::map<std::string, std::shared_ptr<Variant>> m_rt_info;
     };
 
-    template <typename NodeType>
-    class Input
-    {
-    };
-
-    template <typename NodeType>
-    class Output
-    {
-    };
-
     /// \brief A handle for one of a node's inputs.
-    template <>
-    class Input<Node>
+    class NGRAPH_API NodeInput
     {
     public:
-        /// \brief Constructs a Input.
+        /// \brief Constructs a NodeInput.
         /// \param node Pointer to the node for the input handle.
         /// \param index The index of the input.
-        Input(Node* node, size_t index)
+        NodeInput(const Node* node, size_t index)
             : m_node(node)
             , m_index(index)
         {
         }
 
         /// \return A pointer to the node referenced by this input handle.
-        Node* get_node() const { return m_node; }
+        Node* get_node() const { return const_cast<Node*>(m_node); }
```

Comment:
`NodeInput(const_node).get_node()` loses constness.

Reply:
Yes, but in order to keep the original API this is what I have to do. If you want to change the API more I would recommend we do a separate PR. Changing to const breaks other things.